src/ray/gcs/gcs_server/gcs_job_manager.cc

```diff
@@ -203,79 +203,91 @@ void GcsJobManager::HandleGetAllJobInfo(rpc::GetAllJobInfoRequest request,
         job_data_key_to_indices[job_data_key].push_back(i);
       }
 
-      JobID job_id = data.first;
-      WorkerID worker_id = WorkerID::FromBinary(data.second.driver_address().worker_id());
-
-      // If job is not dead, get is_running_tasks from the core worker for the driver.
-      if (data.second.is_dead()) {
-        reply->mutable_job_info_list(i)->set_is_running_tasks(false);
-        core_worker_clients_.Disconnect(worker_id);
+      if (request.has_query_is_running_tasks_field() &&
+          request.query_is_running_tasks_field()) {
```

Comment:
you can omit the has_ check. if it's not set, query_is_running_tasks_field() defaults to false.

Reply:
Good suggestion!